mono/utils/mono-threads-api.h

```diff
@@ -22,35 +22,97 @@ MONO_BEGIN_DECLS
 This API is experimental. It will eventually be required to properly use the rest of the raw-omp embedding API.
 */
 
+typedef struct _MonoStackData {
+	union {
+		int as_int;
+		struct {
+			gboolean has_stackpointer : 1;
```

Comment:
This is quite the overengineering: if stackpointer is NULL then has_stackpointer is FALSE and if stackpointer is non-NULL then has_stackpointer is TRUE, same for function_name and has_function_name, and these 2 `has_*` fields are then used to return NULL or the value (which is itself already NULL or not). Just simplify it and remove this union which is useless.

Otherwise, not-modifying the existing `MONO_API` seems the way to go (as you did it), so looks good to me once you removed this useless union.

Reply:
The boolean meaning false means the struct is not large enough to access the field it refers to. 

That allows safely growing the size of the struct on one side but not the other.

Currently it does not matter, but this addresses your valid concern of updating XA/XM, and then updating runtime again. With the flags in place, we don't have to keep introducing newly-named functions -- instead add flags to the struct (but then the function name isn't great -- "_internal" -- it'll have to be renamed).

Let's see if this somehow fixes the profiler tests too.